### PR TITLE
Find solution in pow list before verify pow from leader

### PR DESCRIPTION
**Pull request by ShengguangXiao**
## Description

Find solution in pow list before verify pow from leader

When DS backup find some different nodes in priority list with same priority, the verify pow ordering will using different nodes, then it cannot find pow in sorted list, it need to check pow in local pow list first, because it is faster, if from local cannot find it, then check the pow from ds leader.

https://github.com/Zilliqa/Issues/issues/477

## Backward Compatibility

- [x] This is not a breaking change
- [ ] This is a breaking change

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [x] local machine test

- [x] small-scale cloud test

### CODE REVIEW

**Comment by bb111189** [comment]
Do we need to add m_mutexAllPOW mutex to RunConsensusOnDSBlockWhenDSPrimary at line 676

File: src/libDirectoryService/DSBlockPreProcessing.cpp [hunk 1]
```diff
@@ -564,6 +527,52 @@ bool DirectoryService::VerifyPoWOrdering(
...
+
+  if (!POW::GetInstance().PoWVerify(m_pendingDSBlock->GetHeader().GetBlockNum(),
+                                    difficulty, headerHash, powSoln.nonce,
+                                    resultStr, mixHashStr)) {
+    LOG_GENERAL(WARNING, "Failed to verify PoW solution from leader for node: "
+                             << pubKey);
+    return false;
+  }
+
+  m_allPoWs[pubKey] = powSoln;
```
Comment (bb111189) [inline src/libDirectoryService/DSBlockPreProcessing.cpp]: let's use at() ?

**Comment by ShengguangXiao** [comment]
> Do we need to add `m_mutexAllPOW` and `m_mutexAllDSPOWs` mutex to RunConsensusOnDSBlockWhenDSPrimary at line 747

I think no need, because when run consensus, we don't receive pow already.

File: src/libDirectoryService/DSBlockPreProcessing.cpp (hunk 1 above)
Comment (ShengguangXiao) [reply src/libDirectoryService/DSBlockPreProcessing.cpp]: This is an insert operation, i will use insert instead

Comment (ShengguangXiao) [reply src/libDirectoryService/DSBlockPreProcessing.cpp]: This is an insert operation, cannot use at

**Pull request merged**